### PR TITLE
tensor.h: Rename cross_product to cross_product_2d/3d

### LINKED ISSUE #1644: regression: cross_product(Tensor&, Tensor)
Due to the recent tensor changes the function
```
void cross_product (Tensor<1,2> &dst,
               const Tensor<1,2> &src);
```
no longer exists breaking applications (like ASPECT). The problem is that the code compiles, but the compiler now picks up
```
Tensor<1,dim> cross_product (const Tensor<1,dim> &src1,
               const Tensor<1,dim> &src2);
```
which fails with an Assert at runtime (dim==3).

Matthias, do you have a suggestion on how we can work around this?

### PR DESCRIPTION
**Pull request by tamiko**
Fixes #1644

### CODE REVIEW

File: include/deal.II/base/tensor_deprecated.h [hunk 1]
```diff
@@ -148,10 +148,24 @@ typename ProductType<Number,OtherNumber>::type
...
+inline
+void
+cross_product (Tensor<1,dim,Number>       &dst,
+               const Tensor<1,dim,Number> &src) DEAL_II_DEPRECATED;
+
 /**
  * The cross-product of 2 vectors in 3d.
  *
- * @deprecated Use the cross_product function that returns the value.
+ * @deprecated Use the function cross_product_3d that returns the value.
```
Comment (tjhei) [inline include/deal.II/base/tensor_deprecated.h]: if you write this as ```cross_product_3d()``` doxygen should link it.

File: include/deal.II/base/tensor_deprecated.h [hunk 2]
```diff
@@ -148,10 +148,24 @@ typename ProductType<Number,OtherNumber>::type
 contract (const Tensor<1,dim,Number> &src1,
           const Tensor<1,dim,OtherNumber> &src2) DEAL_II_DEPRECATED;
 
+
+/**
+ * The cross-product of one vector in 2d. This is just a rotation by 90
+ * degrees.
+ *
+ * @deprecated Use the function cross_product_2d that returns the value.
```
Comment (tjhei) [inline include/deal.II/base/tensor_deprecated.h]: same here

**Comment by tjhei** [comment]
Looks good otherwise!

File: include/deal.II/base/tensor_deprecated.h [hunk 3]
```diff
@@ -148,10 +148,24 @@ typename ProductType<Number,OtherNumber>::type
 contract (const Tensor<1,dim,Number> &src1,
           const Tensor<1,dim,OtherNumber> &src2) DEAL_II_DEPRECATED;
 
+
+/**
+ * The cross-product of one vector in 2d. This is just a rotation by 90
```
Comment (tjhei) [inline include/deal.II/base/tensor_deprecated.h]: I would spell it "cross product".

**Comment by tamiko** [comment]
@tjhei Changed that.

**Pull request merged**